core/src/main/scala/scala/pickling/generator/Macros.scala

```diff
@@ -0,0 +1,100 @@
+package scala.pickling
+package generator
+
+import com.sun.org.apache.xalan.internal.xsltc.compiler.sym
+
+// TODO - move these into Pickler/Unpickler and replace the existing macros.
+object PicklingMacros {
+  import scala.language.experimental.macros
+  def genPickler[T]: Pickler[T] with Generated = macro scala.pickling.generator.Compat.genPickler_impl[T]
```

Comment:
No.  That can actually muck with implicit search.  This was actually the way to get it to work correctly.

I'm not 100% positive we should be annotating this with Generated everywhere, as we aren't using it, but i'm ok keeping it for now.